client/src/app/site/tags/components/tag-list.component.ts

```diff
@@ -0,0 +1,132 @@
+    public editTag = false;
+    public newTag = false;
+    public selectedTag: ViewTag;
+
+    @ViewChild('tagForm')
+    public tagForm: FormGroup;
+
+    /**
+     *
+     * Constructor.
+     * @param titleService
+     * @param translate
+     * @param repo the repository
+     */
+    public constructor(titleService: Title, translate: TranslateService, private repo: TagRepositoryService) {
+        super(titleService, translate);
+    }
+
+    /**
+     * Init function.
+     * Sets the title, inits the table and calls the repo.
+     */
+    public ngOnInit(): void {
+        super.setTitle('Tags');
+        this.initTable();
+        this.tagForm = new FormGroup({ name: new FormControl('', Validators.required) });
+        this.repo.getViewModelListObservable().subscribe(newTags => {
+            this.dataSource.data = newTags;
+        });
+    }
+
+    /**
+     * Sends a new or updates tag to the dataStore
+     */
+    public saveTag(): void {
+        if (this.editTag && this.newTag) {
+            this.submitNewTag();
+        } else if (this.editTag && !this.newTag) {
+            this.submitEditedTag();
+        }
+    }
+
+    /**
+     * Saves a newly created tag.
+     */
+    public submitNewTag(): void {
+        if (this.tagForm.value && this.tagForm.valid) {
+            this.repo.create(this.tagForm.value).subscribe(response => {
+                if (response) {
+                    this.tagForm.reset();
+                    this.cancelEditing();
+                }
+            });
+        }
+    }
+
+    /**
+     * Saves an edited tag.
+     */
+    public submitEditedTag(): void {
+        if (this.tagForm.value && this.tagForm.valid) {
+            const updateData = new Tag({ name: this.tagForm.value.name });
+
+            this.repo.update(updateData, this.selectedTag).subscribe(response => {
+                if (response) {
+                    this.cancelEditing();
+                }
+            });
+        }
+    }
+
+    /**
+     * Deletes the selected Tag
+     */
+    public deleteSelectedTag(): void {
+        this.repo.delete(this.selectedTag).subscribe(response => {
+            this.cancelEditing();
+        });
+    }
+
+    public cancelEditing(): void {
+        this.newTag = false;
+        this.editTag = false;
+        this.tagForm.reset();
+    }
+
+    /**
+     * Select an row in the table
+     * @param tag
+     */
+    public selectTag(viewTag: ViewTag): void {
+        this.selectedTag = viewTag;
+        this.setEditMode(true, false);
+        this.tagForm.setValue({ name: this.selectedTag.name });
+    }
+
+    public setEditMode(mode: boolean, newTag: boolean = true): void {
+        this.editTag = mode;
+        this.newTag = newTag;
+        if (!mode) {
+            this.cancelEditing();
+        }
+    }
+    public keyDownFunction(event: KeyboardEvent): void {
+        if (event.keyCode === 27) {
```

Comment:
this should call your `cancelEditing()` function ;-)
```suggestion
        if (event.keyCode === 27) {
            this.cancelEditing();
        }
```

The form also needs to get resettet. Otherwise you'll end up with the tag name in your form when you press the "plus" button